File: 剑指offer/50.第一个只出现一次的字符.py
from collections import defaultdict


class Solution:
    def firstUniqChar(self, s: str) -> str:
        cnt = defaultdict(int)

        for c in s:
            cnt[c] += 1

        for c in s:
            if cnt[c] == 1:
                return c

        return ' '

File: 剑指offer/50.第一个只出现一次的字符.cpp
#include <string>
#include <unordered_map>
using namespace std;

class Solution {
public:
    char firstUniqChar(string s)
    {
        unordered_map<char, int> cnt;

        for (char c : s) {
            cnt[c] += 1;
        }

        for (char c : s) {
            if (cnt[c] == 1) {
                return c;
            }
        }

        return ' ';
    }
};